src/rgw/rgw_zone_types.h

```diff
@@ -588,22 +631,40 @@ struct RGWZoneGroupPlacementTier {
         decode(t.s3, bl);
       }
     } else if (struct_v >= 3) {
-      if (tier_type == "cloud-s3") {
+      if (is_tier_type_s3()) {
         decode(t.s3, bl);
       }
       decode(allow_read_through, bl);
       decode(read_through_restore_days, bl);
     }
+    if (struct_v >= 4) {
+      decode(restore_storage_class, bl);
+      if (is_tier_type_s3_glacier()) {
+        decode(s3_glacier, bl);
+      }
+    }
     DECODE_FINISH(bl);
   }
 
+  bool is_tier_type_s3() const {
+    return (tier_type == "cloud-s3" || tier_type == "cloud-s3-glacier");
+  }
+
+  bool is_tier_type_s3_glacier() const {
+    return (tier_type == "cloud-s3-glacier");
+  }
+
   void dump(Formatter *f) const;
   void decode_json(JSONObj *obj);
   static void generate_test_instances(std::list<RGWZoneGroupPlacementTier*>& o) {
     o.push_back(new RGWZoneGroupPlacementTier);
     o.push_back(new RGWZoneGroupPlacementTier);
     o.back()->tier_type = "cloud-s3";
     o.back()->storage_class = "STANDARD";
+    o.back()->allow_read_through = false;
+    o.back()->restore_storage_class = "STANDARD";
```

Comment:
Since you have a macro RGW_STORAGE_CLASS_STANDARD for this, maybe use it instead of hard-coding the string here and a few lines above. (I realize you're just duplicating the local coding style.)

Reply:
Done.